app/controllers/admin/talentlms_controller.rb

```diff
@@ -7,28 +7,57 @@
 module Admin
   class TalentlmsController < ApplicationController
     before_action :require_can_manage_config!
+    before_action :set_config, only: [:index, :edit, :update]
 
     def index
-      @config = Talentlms::Config.first_or_initialize
+    end
+
+    def new
+      @config = config_scope.first_or_initialize
+      # If a record already exists, we want to be editing. This redirect will
+      # send the user to edit the existing record if one is found.
+      # If we ever have the need for multiple records at once, this will need to be adjusted.
+      redirect_to edit_admin_talentlm_path(@config) unless @config.new_record?
     end
 
     def create
-      @config = Talentlms::Config.first_or_initialize
-      @config.assign_attributes(config_params)
-      if config.valid?
-        @config.save
-        redirect_to action: :index
+      @config = config_scope.new
+      if @config.update(config_params)
+        respond_with(@config, location: edit_admin_talentlm_path(@config))
       else
-        render :index
+        render 'new'
       end
     end
 
+    def edit
+    end
+
+    def update
+      @config.update(config_params)
+      respond_with(@config, location: edit_admin_talentlm_path(@config))
+    end
+
+    private def config_scope
+      Talentlms::Config
+    end
+
     def config_params
       params.require(:talentlms_config).permit(
         :subdomain,
         :api_key,
         :courseid,
       )
     end
+
+    private def set_config
+      @config = config_scope.first_or_initialize
+
+      # Checks to make sure we are working with the expected record in the database.
+      # If a record already exists, we want to be editing that record. If one does
+      # not exist, we want to be creating it.
+      # If we ever have the need for multiple records at once, this will need to be adjusted.
+      redirect_to new_admin_talentlm_path if @config.new_record?
+      redirect_to edit_admin_talentlm_path(@config) if @config.id && @config.id != params[:id].to_i
```

Comment:
This seems overly complicated, since `set_config` is really simple, and basically already integrated in `new` maybe we should just throw it away and add `@config = config_scope.first_or_initialize` to the top of each `index`, `edit` and `update`

Reply:
Looks like maybe we just went a redirect in `index` too to send you to new or edit for now.

Reply:
That all makes sense and I can definitely run through this again. I wasn't sure how complicated we wanted the logic to get with the expectation that there's only one record in the database. For example if the new action is called when a record already exists, should that be recognized and redirected to an edit? Similarly with the edit if the wrong id is passed.